Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/cryptography/AesCryptography.kt

```diff
@@ -1,16 +1,37 @@
 package de.rki.coronawarnapp.covidcertificate.cryptography
 
+import com.google.android.gms.common.util.Hex
 import dagger.Reusable
-import okio.ByteString
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.util.encoders.Base64.decode
+import java.security.Security
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
 import javax.inject.Inject
 
 @Reusable
 class AesCryptography @Inject constructor() {
 
+    private val ivParameterSpec
+        get() = IvParameterSpec(Hex.stringToBytes("00000000000000000000000000000000"))
+
     fun decrypt(
         decryptionKey: ByteArray,
-        encryptedData: ByteString
+        encryptedData: ByteArray
     ): ByteArray {
-        throw NotImplementedError()
+        Security.addProvider(BouncyCastleProvider())
```

Comment:
Could it become a problem if we add it multiple times (this could happen because of `@Reusable`)?

Should we move it to `CoronaWarnApplication` as it affects the whole app once called? We already have  this there:

```kotlin
Security.insertProviderAt(Conscrypt.newProvider(), 1)
```

Or we clean it up a bit and introduce a `CryptographySetup.setup()` class that we can manually call during unit tests and during `App.onCreate()`.
Is it possible to include a `teardown()` so we can clean up after the unit tests too?

Reply:
Seems like this is not necessary at all.